Bt security api renaming

joerchan opened the pull request:
Update security API naming.

joerchan committed:
Bluetooth: Host: Rename security level enum

Rename security level enum, using level and number instead of low,
medium, high and fips.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
Bluetooth: Host: Rename API function to initiate bluetooth security.

Rename bt_conn_security to bt_conn_set_security, this makes the API
naming more consistent.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

carlescufi commented on include/bluetooth/conn.h:
```diff
@@ -279,15 +279,22 @@ struct bt_conn *bt_conn_create_slave_le(const bt_addr_le_t *peer,
 	/** Only for BR/EDR special cases, like SDP */
-	BT_SECURITY_NONE,
+	BT_SECURITY_L0,
 	/** No encryption and no authentication. */
```
Can you add: `Level 1: No encryption and no authentication`
same with the rest

joerchan replied:
Done.

carlescufi reviewed (changes requested):
Nice, thank you! Tiny change requested

joerchan committed:
Bluetooth: Host: Rename long error codes

Rename AUTHENTICATION, to AUTH, since this is a well established short
form of the word.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

Vudentz reviewed (approved):
LGTM
